python/ray/serve/backend_worker.py

```diff
@@ -256,7 +257,39 @@ async def invoke_single(self, request_item: Query) -> Any:
 
         start = time.time()
         try:
-            result = await method_to_call(arg)
+            if self.config.internal_metadata.is_asgi_app:
```

Comment:
Please pull this logic out and add unit tests for it

Reply:
it's a bit hard to pull out, I think we should do it after remove the batching part so there's only one "invoke", and then we can make it invoke & invoke_asgi

Reply:
sure, sounds good. can you leave a note so we remember to clean it up when that happens?